### PR TITLE
[AUD-1620] Add ListeningHistoryScreen and update TrackList component to properly play and display tracks

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description

* Add ListeningHistoryScreen
* Update TrackList component
* Split out TrackArtwork to its own component
* Removed the bottom margin/padding from a lot of components where it is no longer needed

[AUD-1620]

Additional Notes:
* Will add screenshots
* Will remove the ios file update
* May do more refactoring of the TrackList later bc there was a lot of duplicated code to marshal the track data

### Dragons

N/A

### How Has This Been Tested?

Lots of manual testing

### How will this change be monitored?

N/A

### CODE REVIEW

File: packages/mobile/src/assets/images/pbIconPauseAlt.svg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="32px" height="32px" viewBox="0 0 32 32" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <g id="Assets" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g id="assets" transform="translate(-847.000000, -457.000000)">
+            <circle id="Oval" fill="#00000000" cx="863" cy="473" r="12"></circle>
+            <path d="M863,457 C871.836556,457 879,464.163444 879,473 C879,481.836556 871.836556,489 863,489 C854.163444,489 847,481.836556 847,473 C847,464.163444 854.163444,457 863,457 Z M860,467 L858,467 C857.447715,467 857,467.447715 857,468 L857,468 L857,478 C857,478.552285 857.447715,479 858,479 L858,479 L860,479 C860.552285,479 861,478.552285 861,478 L861,478 L861,468 C861,467.447715 860.552285,467 860,467 L860,467 Z M868,467 L866,467 C865.447715,467 865,467.447715 865,468 L865,468 L865,478 C865,478.552285 865.447715,479 866,479 L866,479 L868,479 C868.552285,479 869,478.552285 869,478 L869,478 L869,468 C869,467.447715 868.552285,467 868,467 L868,467 Z" id="Combined-Shape" fill="#FFFFFFBF"></path>
```
Comment (dylanjeffers) [inline packages/mobile/src/assets/images/pbIconPauseAlt.svg:6]: should we change the fill to #000 so we can pass fill prop to it?

File: packages/mobile/src/assets/images/pbIconPlay.svg [hunk 2]
```diff
@@ -2,8 +2,8 @@
 <svg width="32px" height="32px" viewBox="0 0 32 32" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
     <g id="Assets" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
         <g id="assets" transform="translate(-847.000000, -503.000000)">
-            <circle id="Oval" fill="#000" cx="863" cy="519" r="12"></circle>
-            <g id="icon/playbar/play" transform="translate(847.000000, 503.000000)" fill="#f00">
+            <circle id="Oval" fill="#FFFFFF" cx="863" cy="519" r="12"></circle>
+            <g id="icon/playbar/play" transform="translate(847.000000, 503.000000)" fill="#CC0FE0">
```
Comment (dylanjeffers) [inline packages/mobile/src/assets/images/pbIconPlay.svg:6]: and i think same here?

File: packages/mobile/src/components/app-navigator/BottomTabNavigator.tsx [hunk 3]
```diff
@@ -93,6 +94,10 @@ const ProfileStackScreen = createStackScreen<ProfileStackParamList>(Stack => (
     <Stack.Screen name='EditProfile' component={EditProfileScreen} />
     <Stack.Screen name='SettingsScreen' component={SettingsScreen} />
     <Stack.Screen name='AboutScreen' component={AboutScreen} />
+    <Stack.Screen
+      name='ListeningHistoryScreen'
```
Comment (dylanjeffers) [inline packages/mobile/src/components/app-navigator/BottomTabNavigator.tsx:98]: ah these screens have incorrect names above, standard is to not put "Screen" in the name, can maybe address later

File: packages/mobile/src/components/track-list/TrackListItem.tsx [hunk 4]
```diff
@@ -291,12 +210,19 @@ export const TrackListItem = ({
...
           <IconButton
             icon={IconHeart}
-            styles={{ icon: { height: 16, width: 16 } }}
+            styles={tileIconStyles}
             fill={isSaved ? themeColors.primary : themeColors.neutralLight4}
-            onPress={onSaveTrack}
+            onPress={handleSaveTrack}
+          />
+        )}
+        {handleClickOverflow && trackItemAction === 'overflow' && (
```
Comment (dylanjeffers) [inline packages/mobile/src/components/track-list/TrackListItem.tsx:221]: tern

File: packages/mobile/src/components/track-list/TrackListItem.tsx [hunk 5]
```diff
@@ -291,12 +210,19 @@ export const TrackListItem = ({
             <UserBadges user={user} badgeSize={12} hideName />
           </Text>
         </View>
-        {onSaveTrack && trackItemAction === 'save' && (
+        {handleSaveTrack && trackItemAction === 'save' && (
```
Comment (dylanjeffers) [inline packages/mobile/src/components/track-list/TrackListItem.tsx:213]: also same!

File: packages/mobile/src/screens/audio-screen/AudioScreen.tsx [hunk 6]
```diff
@@ -57,10 +57,7 @@ const useStyles = makeStyles(({ spacing, palette, typography }) => ({
     display: 'flex',
     height: screenHeight
   },
-  container: {
-    // TODO: Need to fix this spacing issue on the bottom of pages to account for the bottom bar and now playlist drawer
-    marginBottom: 240
-  },
+  container: {},
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/audio-screen/AudioScreen.tsx:60]: can we remove?

File: packages/web/src/pages/history-page/store/lineups/tracks/sagas.js [hunk 7]
```diff
@@ -58,7 +58,8 @@ class TracksSagas extends LineupSagas {
     super(
       PREFIX,
       tracksActions,
-      store => store.history.tracks,
+      // store => store.history.tracks,
```
Comment (dylanjeffers) [inline packages/web/src/pages/history-page/store/lineups/tracks/sagas.js:61]: comment

File: packages/mobile/src/screens/settings-screen/ListeningHistoryScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    user: artists[track.owner_id]
+  }))
+
+  const isQueued = () => {
+    return tracksWithUsers.some((track: any) => playingUid === track.uid)
+  }
+
+  const queuedAndPlaying = isPlaying && isQueued
+
+  const trackList: ListTrackMetadata[] = tracksWithUsers
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/settings-screen/ListeningHistoryScreen.tsx:84]: not sure of overall number of re-renders for this page, but if it seems like a lot, potentially we can figure out how to memoize various things if needed?

**Review by dylanjeffers — APPROVED** [review]
looks really clean! only callout is the color changes in the icons, whether we are cool with hard-coding some colors, or whether we should always pass fill prop?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Add-ListeningHistoryScreen-to-settings-page

File: packages/mobile/package-lock.json [hunk 9]
```diff
@@ -5755,8 +5755,6 @@
     "audius-client": {
       "version": "file:../web",
       "requires": {
-        "@audius/libs": "1.2.85",
```
Comment (sliptype) [inline packages/mobile/package-lock.json:5758]: Strange, why would audius-client no longer have these dependencies?

File: packages/mobile/src/screens/explore-screen/tabs/ForYouTab/ForYouTab.tsx [hunk 10]
```diff
@@ -35,9 +35,7 @@ const createStyles = (themeColors: ThemeColors) =>
     contentContainer: {
       display: 'flex',
       padding: 12,
-      paddingVertical: 24,
-      // TODO: Fix this
-      marginBottom: 240
```
Comment (sliptype) [inline packages/mobile/src/screens/explore-screen/tabs/ForYouTab/ForYouTab.tsx:40]: How did this get fixed? I had to do the same thing on collections

**Review by sliptype — APPROVED** [review]
Good stuff! Looking forward to not having to map all the track data before passing to TrackList, that will be dope

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Add-ListeningHistoryScreen-to-settings-page

File: packages/mobile/src/screens/settings-screen/ListeningHistoryScreen.tsx (hunk 8 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/settings-screen/ListeningHistoryScreen.tsx:81]: just checked, its fine!

**Pull request merged**